File: Template/Architecture/MVC.xctemplate/___FILEBASENAME___/Controller/___FILEBASENAME___ViewController.m
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//  ___COPYRIGHT___
//

#import "___FILEBASENAME___ViewController.h"

@interface ___FILEBASENAMEASIDENTIFIER___ViewController ()

@end

@implementation ___FILEBASENAMEASIDENTIFIER___ViewController

// 把控制器导出为组件
XF_EXPORT_COMPONENT

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    // 配置当前视图
    [self config];
    // 初始化视图
    [self setUpViews];
}

#pragma mark - 初始化
- (void)config {
    
}

- (void)setUpViews {
    
}


#pragma mark - LifeCycle
// 接收到组件返回数据
- (void)onNewIntent:(id)intentData
{
    
}

// 注册MVx通知
- (void)registerMVxNotifactions
{
    // 注册MVx构架通知
    //    XF_RegisterMVxNotis_(XFUserDidLoginNotifaction)
}


// 接受到MVx构架通知或XFLegoVIPER模块的事件
- (void)receiveComponentEventName:(NSString *)eventName intentData:(id)intentData
{
    // 匹配对应通知
    /*XF_EventIs_(XFUserDidLoginNotifaction, {
     // TODO
     })*/
}


#pragma mark - Change UI Action


#pragma mark - UIControlDelegate


#pragma mark - Getter

@end

File: Template/Architecture/MVVM@Swift.xctemplate/___FILEBASENAME___/View/___FILEBASENAME___ViewController.swift
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//  ___COPYRIGHT___
//

import UIKit
import XFLegoVIPER

private func DataDriver(_ viewController: UIViewController) ->___VARIABLE_productName:identifier___DataDriverProtocol? { return viewController.dataDriver as? ___VARIABLE_productName:identifier___DataDriverProtocol }

class ___VARIABLE_productName:identifier___ViewController: UIViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        setUpUI()
    }
    
    //MARK:- Getter
    
}

//MARK:- init UI
extension ___VARIABLE_productName:identifier___ViewController {
    
    fileprivate func setUpUI() {
        config()
        setUpViews()
        bindViewData()
    }
    
    private func config() {
        
    }
    
    private func setUpViews() {
        
    }
    
    private func bindViewData() {
        // 调用ViewModel方法
        //        DataDriver(self)?.fetchData()
    }
}

//MARK:- Change UI State
extension ___VARIABLE_productName:identifier___ViewController : ___VARIABLE_productName:identifier___ViewProtocol {
    
}

//MARK:- DataSource
extension ___VARIABLE_productName:identifier___ViewController {
    
}

//MARK:- UIControlDelegate
extension ___VARIABLE_productName:identifier___ViewController {
    
}
